pkg/cloud/ibm/assets/deployment.yaml

```diff
@@ -0,0 +1,132 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: ibm-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+  labels:
+    k8s-app: ibm-cloud-controller-manager
+spec:
+  replicas: 2
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 2
+  selector:
+    matchLabels:
+      k8s-app: ibm-cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        k8s-app: ibm-cloud-controller-manager
+    spec:
+      priorityClassName: system-cluster-critical
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - topologyKey: "kubernetes.io/hostname"
+            labelSelector:
+              matchLabels:
+                k8s-app: ibm-cloud-controller-manager
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+        operator: Exists
+      - effect: NoExecute
+        key: node.kubernetes.io/unreachable
+        operator: Exists
+        tolerationSeconds: 120
+      - effect: NoExecute
+        key: node.kubernetes.io/not-ready
+        operator: Exists
+        tolerationSeconds: 120
+      - effect: NoSchedule
+        key: node.cloudprovider.kubernetes.io/uninitialized
+        operator: Exists
+      - effect: NoSchedule
+        key: node.kubernetes.io/not-ready
+        operator: Exists
+      serviceAccountName: cloud-controller-manager
+      terminationGracePeriodSeconds: 90
+      containers:
+      # Container is required to specify name in order for image substitution
+      # to take place. Cloud-controller-manager is substituted with
+      # real CCM image from the cluster
+      - name: cloud-controller-manager
+        image: quay.io/openshift/origin-ibm-cloud-controller-manager
+        imagePullPolicy: IfNotPresent
+        securityContext:
```

Comment:
i think we will need to drop this securityContext. the other clouds do not have this and @cjschaef is seeing some root-related issues when running the ccm.

Reply:
I'm will be testing using the [IBM CCM Dockerfile.openshift](https://github.com/openshift/cloud-provider-ibm/blob/master/openshift-hack/images/Dockerfile.openshift) and not limiting the container to `runAsNonRoot` to check if things work as expected.

Reply:
i have a feeling that the `runAsNonRoot` might be causing issues because we do not specify a `runAsUser`. the product docs say this:

> MustRunAsNonRoot - Requires that the pod be submitted with a non-zero runAsUser or have the USER directive defined in the image. No default provided.

ref: https://docs.openshift.com/container-platform/4.8/authentication/managing-security-context-constraints.html#authorization-SCC-strategies_configuring-internal-oauth

Reply:
also, looking at the image layers it appears that we only define user `0`. so, i imagine this context is forcing the check, which then fails. i'm still digging to learn a little more about how we specify the user.

Reply:
securityContext removed from deployment.yaml